src/librustc_mir/shim.rs

```diff
@@ -259,6 +272,164 @@ impl<'a, 'tcx> DropElaborator<'a, 'tcx> for DropShimElaborator<'a, 'tcx> {
     }
 }
 
+/// Build a `Clone::clone` shim for `recvr_ty`. Here, `def_id` is `Clone::clone`.
+fn build_clone_shim<'a, 'tcx>(tcx: ty::TyCtxt<'a, 'tcx, 'tcx>,
+                              def_id: DefId,
+                              rcvr_ty: ty::Ty<'tcx>)
```

Comment:
don't call this `rcvr_ty` - `ty` is fine. The *receiver* is the first argument of the method, which has type `&Self` (e.g. `&[Vec<u32>; 2]` if the impl is for `<[Vec<u32>; 2] as Clone>`. Also fix the comment above.